pkg/registry/apis/dashboard/legacy/migrate.go

```diff
@@ -0,0 +1,289 @@
+package legacy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/grpc/metadata"
+
+	"github.com/grafana/authlib/claims"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	dashboard "github.com/grafana/grafana/pkg/apis/dashboard"
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+)
+
+type migrator = func(ctx context.Context, orgId int64, opts MigrateOptions, stream resource.ResourceStore_BatchProcessClient) error
+
+func (a *dashboardSqlAccess) Migrate(ctx context.Context, opts MigrateOptions) (*resource.BatchResponse, error) {
+	info, err := claims.ParseNamespace(opts.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// Migrate everything
+	if len(opts.Resources) == 1 && opts.Resources[0] == "*" {
+		opts.SendHistory = true
+		opts.Resources = []string{
+			"folders",
+			"dashboards",
+			"panels",
+		}
+	}
+
+	migrators := []migrator{}
+	settings := resource.BatchSettings{
+		RebuildCollection: true,
+		SkipValidation:    true,
+	}
+
+	for _, res := range opts.Resources {
+		switch res {
+		case "folders":
+			migrators = append(migrators, a.migrateFolders)
+			settings.Collection = append(settings.Collection, &resource.ResourceKey{
+				Namespace: opts.Namespace,
+				Group:     "folder.grafana.app",
```

Comment:
nit.
This could be imported as well https://github.com/grafana/grafana/blob/bc6b7a4df62224890ae419895f0ed78056b6e374/pkg/apis/folder/v0alpha1/register.go#L13